File: Lesson3/Unit3.pas
unit Unit3;

interface

uses
  System.SysUtils, System.Types, System.UITypes, System.Classes, System.Variants,
  FMX.Types, FMX.Controls, FMX.Forms, FMX.Graphics, FMX.Dialogs, FMX.StdCtrls,
  FMX.Layouts, FMX.Controls.Presentation, FMX.Edit, Data.Bind.EngExt,
  Fmx.Bind.DBEngExt, System.Rtti, System.Bindings.Outputs, Fmx.Bind.Editors,
  Data.Bind.Components;

type
  TForm1 = class(TForm)
    Timer1: TTimer;
    Edit1: TEdit;
    Layout1: TLayout;
    Button1: TButton;
    Button2: TButton;
    Button3: TButton;
    Switch1: TSwitch;
    BindingsList1: TBindingsList;
    LinkControlToPropertyEnabled: TLinkControlToProperty;
    procedure Button3Click(Sender: TObject);
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
  private
    { private êÈåæ }
  public
    { public êÈåæ }
  end;

var
  Form1: TForm1;

implementation

{$R *.fmx}

procedure TForm1.Button1Click(Sender: TObject);
begin
  Timer1.interval := 1000;

end;

procedure TForm1.Button2Click(Sender: TObject);
begin
  Timer1.interval := 2000;
end;

procedure TForm1.Button3Click(Sender: TObject);
begin
  Timer1.interval := 500;
end;

end.

File: Lesson3/CPP/Unit3.cpp
//---------------------------------------------------------------------------

#include <fmx.h>
#pragma hdrstop

#include "Unit3.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.fmx"
TForm2 *Form2;
//---------------------------------------------------------------------------
__fastcall TForm2::TForm2(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm2::Button3Click(TObject *Sender)
{
	Memo1->Lines->Insert(0,L"インターバルを 500 に変更します");
	Timer1->Interval = 500;
}
//---------------------------------------------------------------------------
void __fastcall TForm2::Button1Click(TObject *Sender)
{
	Memo1->Lines->Insert(0,L"インターバルを 1000 に変更します");
	Timer1->Interval = 1000;
}
//---------------------------------------------------------------------------
void __fastcall TForm2::Button2Click(TObject *Sender)
{
	Memo1->Lines->Insert(0,L"インターバルを 2000 に変更します");
	Timer1->Interval = 2000;
}
//---------------------------------------------------------------------------

void __fastcall TForm2::Timer1Timer(TObject *Sender)
{
	// 現在の日付時刻は Now() で取得できます。
	// 取得した値は TDateTime 型であり、これはそのまま表示できないため、
	// DateTimetoStr() という関数で文字列型に変換しています。
	Memo1->Lines->Insert(0,DateTimeToStr(Now()));
}
//---------------------------------------------------------------------------

